File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/Finance.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class Finance

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _pension As Decimal
    <DisplayName("Pension"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property pension() As Decimal
        Get
            Return _pension
        End Get
        Set(ByVal value As Decimal)
            _pension = value
            totalFinance = pension + savings
        End Set
    End Property

    Private _savings As Decimal
    <DisplayName("Regular Savings"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property savings() As Decimal
        Get
            Return _savings
        End Get
        Set(ByVal value As Decimal)
            _savings = value
            totalFinance = pension + savings
        End Set
    End Property

    Private _totalFinance As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Finance Outgoings")> _
    Public Property totalFinance() As Decimal
        Get
            Return _totalFinance
        End Get
        Set(ByVal value As Decimal)
            _totalFinance = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Finance", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalFinance.ToString("c2")
    End Function

End Class

File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/Finance.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;


namespace Household_budget_calculator_CS.expanding_property_classes
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class Finance
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _pension;
        [DisplayName("Pension"), RefreshProperties(RefreshProperties.All)]
        public decimal pension
        {
            get { return _pension; }
            set
            {
                _pension = value;
                totalFinance = pension + savings;
            }
        }

        private decimal _savings;
        [DisplayName("Regular Savings"), RefreshProperties(RefreshProperties.All)]
        public decimal savings
        {
            get { return _savings; }
            set
            {
                _savings = value;
                totalFinance = pension + savings;
            }
        }

        private decimal _totalFinance;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Finance Outgoings")]
        public decimal totalFinance
        {
            get { return _totalFinance; }
            set
            {
                _totalFinance = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Finance",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalFinance.ToString("c2");
        }

    }
}
